pyccel/ast/operators.py

```diff
@@ -761,14 +813,12 @@ class PyccelBooleanOperator(PyccelOperator):
     arg2: PyccelAstNode
         The second argument passed to the operator
     """
-    _dtype = NativeBool()
-    _rank  = 0
-    _shape = ()
-    _precision = default_precision['bool']
 
-    def _set_dtype(self):
+    @staticmethod
+    def _calculate_dtype():
```

Comment:
This function is called from `PyccelOperator.__init__`, therefore all the different versions of this function need to have the same prototype. Otherwise there is no way to know what arguments need to be passed

Reply:
ow right. Thank you Very much